bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -1225,7 +1241,7 @@ public void operationComplete(Future<Channel> future) throws Exception {
                         if (future.isSuccess() && state == ConnectionState.CONNECTING) {
                             LOG.error("Connection state changed before TLS handshake completed {}/{}", addr, state);
                             rc = BKException.Code.BookieHandleNotAvailableException;
-                            closeChannel(future.get());
```

Comment:
Why need this change?

Reply:
In the error scenario where there is a key mismatch, future obj is null.